### DISCUSSION TITLE
Saving memory-only chart as PNG

### DISCUSSION THREAD

**Discussion by hellb0rg** [Live-Charts/LiveCharts2#148] [Q&A] [answered]
i tried looking around but was unable to find any answer.
I am creating a chart which exists only in memory and is never shown in the UI. At one point i would like to save this chart as a png file, but i cant seem to get this working.

I went looking at the documentation and samples for livecharts1.0 - WPF, which gives this code:
```C#
var myChart = new LiveCharts.Wpf.CartesianChart
{
    DisableAnimations = true,
    Width = 600,
    Height = 200,
    Series = new SeriesCollection
    {
        new LineSeries
        {
            Values = new ChartValues<double> {1, 6, 7, 2, 9, 3, 6, 5}
        }
    }
};
var viewbox = new Viewbox();
viewbox.Child = myChart;
viewbox.Measure(myChart.RenderSize);
viewbox.Arrange(new Rect(new Point(0, 0), myChart.RenderSize));
myChart.Update(true, true); //force chart redraw
viewbox.UpdateLayout();

SaveToPng(myChart, "chart.png");
```

But myChart doesnt contain the function Update(bool, bool) and if it try to run the code i get a System.NullReferenceException on this row: 
```C# 
viewbox.Arrange(new Rect(new Point(0, 0), myChart.RenderSize));
```
If someone could at least point me in the right direction, id be ever so happy.

**Comment by ghost** [comment, marked as answer]
Use SKCartesianChart to save the chart as a PNG file:
```
var cartesianChart = new SKCartesianChart
{
    Width = 600,
    Height = 200,
    Series = new ObservableCollection<ISeries>
    {
        new LineSeries<double>
        {
            Values = new ObservableCollection<double> { 1, 6, 7, 2, 9, 3, 6, 5 }
        }
    }
};
cartesianChart.SaveImage("chart.png");
```

**Reply by hellb0rg** [reply]
That worked perfect. Thanks a bunch!
